src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -261,7 +261,14 @@ public static string SelectProductNameForDirectory(Platform.XDG_Type dirpath)
                         // create the xdg folder if needed
                         if (!Directory.Exists(xdgDataHomeDefault))
                         {
-                            Directory.CreateDirectory(xdgDataHomeDefault);
+                            try
+                            {
+                                Directory.CreateDirectory(xdgDataHomeDefault);
+                            }
+                            catch (UnauthorizedAccessException)
+                            {
+                                //service accounts won't have permission to create user folder
```

Comment:
Path.Combine() is used often with this api and would have a problem with null.  My thinking was that this api returns the expected path which may be used in error message.